### DISCUSSION TITLE
Mesh element

### DISCUSSION THREAD

**Discussion by Nikhil23121998** [peridigm/peridigm#328] [Q&A]
Is there any limit to mesh elements while doing simulation? When I am using fine meshing, it shows some errors.

**Comment by oldninja** [comment]
I don't think there is a limit. How many cells do you have?

**Reply by djlittl** [reply]
I have run simulations with tens of millions of elements, but eventually you run into integer overflow errors and/or you run out of memory.  It’s the number of bonds that causes problems, since there are typically several hundred bonds attached to each node.  You should check to make sure your horizon is set to a sane value.  If you set it to a large value relative to the mesh size, you can run out of memory.  If you are refining the mesh, be aware that if you don’t reduce the horizon at the same time, the number of bonds will go up very quickly.

Different topic:  I have had trouble with the text file discretization file format for large meshes.  If you are using a text file for the mesh, you may want to convert it to an exodus/genesis file using the text_to_exodus.py script in the Peridigm scripts directory.
